semgrep/semgrep/cli.py

```diff
@@ -233,48 +232,43 @@ def cli() -> None:
     # change cwd if using docker
     semgrep.config_resolver.adjust_for_docker(args.precommit)
 
-    try:
-        if args.dump_ast:
-            if not args.lang:
-                print_error_exit("language must be specified to dump ASTs")
-            else:
-                dump_parsed_ast(args.json, args.lang, args.pattern, args.target)
-        elif args.validate:
-            _, invalid_configs = semgrep.semgrep_main.get_config(
-                args.generate_config, args.pattern, args.lang, args.config
-            )
-            if invalid_configs:
-                print_error_exit(
-                    f"run with --validate and there were {len(invalid_configs)} errors loading configs"
-                )
-            else:
-                print_error("Config is valid")
-
-        elif args.test:
-            semgrep.test.test_main(args)
+    if args.dump_ast:
+        if not args.lang:
+            print_error_exit("language must be specified to dump ASTs")
         else:
-            semgrep.semgrep_main.main(
-                target=args.target,
-                pattern=args.pattern,
-                lang=args.lang,
-                config=args.config,
-                generate_config=args.generate_config,
-                no_rewrite_rule_ids=args.no_rewrite_rule_ids,
-                jobs=args.jobs,
-                include=args.include,
-                include_dir=args.include_dir,
-                exclude=args.exclude,
-                exclude_dir=args.exclude_dir,
-                json_format=args.json,
-                sarif=args.sarif,
-                output_destination=args.output,
-                quiet=args.quiet,
-                strict=args.strict,
-                exit_on_error=args.error,
-                autofix=args.autofix,
-                dangerously_allow_arbitrary_code_execution_from_rules=args.dangerously_allow_arbitrary_code_execution_from_rules,
+            dump_parsed_ast(args.json, args.lang, args.pattern, args.target)
+    elif args.validate:
+        _, invalid_configs = semgrep.semgrep_main.get_config(
+            args.generate_config, args.pattern, args.lang, args.config
+        )
+        if invalid_configs:
+            print_error_exit(
+                f"run with --validate and there were {len(invalid_configs)} errors loading configs"
             )
-    except NotImplementedError as ex:
-        print_error_exit(
-            f"semgrep encountered an error: {ex}; this is not your fault. {PLEASE_FILE_ISSUE_TEXT}"
```

Comment:
I don't think this error message was very useful, an error here could have been there fault. For example, if they use an unknown pattern operator it _is their fault_ and they probably shouldn't file an issue in this situation:

```
test.yaml has invalid rule key {'pattern-unknown'} at rule id test-parent, can only have: {'metadata', 'fix', 'patterns', 'pattern', 'severity', 'equivalences', 'id', 'languages', 'pattern-regex', 'paths', 'pattern-either', 'message'}
no valid configuration file found (1 configs were invalid)
```

Note this case is now being handled in `__main__`.